src/java.base/share/classes/com/sun/crypto/provider/BlockCipherParamsCore.java

```diff
@@ -48,8 +49,11 @@
     private int block_size = 0;
     private byte[] iv = null;
 
-    BlockCipherParamsCore(int blksize) {
+    private int[] moreSizes = null;
```

Comment:
The moreSizes is not used other than the init() method field.  It may be not easy to check the specific size if we cache all supported sized in the object.  For example, if the required IV size if 8 bytes, it may be a problem about how to make sure the iv size is 8 bytes exactly for a specific algorithm.

Maybe, we could just have a ivSize filed.  The default value is block_size, which coupe be set with the init(ivParameterSpec) method.

```
    ....
    private int ivSize;
    ...
   BlockCipherParamsCore(int blkSize) {
       block_size = blkSize;
       ivSize = blkSize;
    }
    ...
   void init(AlgorithmParameterSpec paramSpec) {
        ivSize = ...;  // reset IV size.
    }

    // use ivSize while encoding and decoding.

```

Reply:
The problem with this is that this assumes that the specified paramSpec argument of the init() method is always valid.